src/components/ViewResourceModal.jsx

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Stanford University see LICENSE for license
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import { useDispatch, useSelector } from 'react-redux'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faCopy, faEdit } from '@fortawesome/free-solid-svg-icons'
+import ModalWrapper, { useDisplayStyle, useModalCss } from 'components/ModalWrapper'
+import { hideModal } from 'actions/modals'
+import { selectModalType } from 'selectors/modals'
+import { selectCurrentResource } from 'selectors/resources'
+import ResourceComponent from './editor/ResourceComponent'
+
+const ViewResourceModal = (props) => {
+  const dispatch = useDispatch()
+
+  const show = useSelector((state) => selectModalType(state) === 'ViewResourceModal')
+
+  // Ensure there is a current resource before attempting to render a resource component
+  const currentResource = useSelector((state) => selectCurrentResource(state))
+
+  const close = (event) => {
+    event.preventDefault()
+    dispatch(hideModal())
+  }
+
+  const editAndClose = (event) => {
+    close(event)
+    props.handleEdit(currentResource.uri)
+  }
+
+  const copyAndClose = (event) => {
+    close(event)
+    props.handleCopy(currentResource.uri)
+  }
+
+  const modal = (
+    <div className={ useModalCss(show) }
+         tabIndex="-1"
+         role="dialog"
+         id="view-resource-modal"
+         data-testid="view-resource-modal"
+         style={{ display: useDisplayStyle(show) }}>
+      <div className="modal-dialog modal-xl modal-dialog-scrollable" role="document">
+        <div className="modal-content">
+          <div className="modal-header">
+            <h4 className="modal-title">View Resource</h4>
+            <div className="view-resource-buttons">
+              <button type="button" className="close" onClick={ close } aria-label="Close">
+                <span aria-hidden="true">&times;</span>
+              </button>
+            </div>
+          </div>
+          <div className="modal-body view-resource-modal-content">
+            { currentResource && <ResourceComponent />}
```

Comment:
I think that rather than `current resource &&` to handle when no resource, around L73 should be: `if(!resource) return null`.

Reply:
The complication is that this line will still error out even if we return null beneath it. And we can't add the check before this point, or the app will complain about using React Hooks with a conditional.